### PR TITLE
Toolchain build fix

### PR DESCRIPTION
**Pull request by CarlWachter**
This is very WIP

### CODE REVIEW

**Comment by CarlWachter** [comment]
CI runs (aside from the fact that it can't publish due to me not having the rights)

After hermit-os/hermit-playground#30 and hermit-os/newlib#2, the clones in toolchain.sh can be changed back to the hermit-os branches. Then this can be merged

**Comment by mkroening** [comment]
@CarlWachter, the other PRs are merged. :)

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
     branches:
       - master
   workflow_dispatch:
+  pull_request:
```
Comment (mkroening) [inline .github/workflows/publish.yml:8]: We can keep this, but please move it before `push`.

**Comment by CarlWachter** [comment]
Cleaned up.

This should be functional now, when executed with publishing rights, i.e. ready to merge, once the CI has failed at the right point.

**Pull request merged**